tests/LogNonGenuineHostTest.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace MongoDB\Tests;
+
+use MongoDB\Client;
+use MongoDB\Driver\Exception\InvalidArgumentException;
+use Psr\Log\AbstractLogger;
+use Psr\Log\LoggerInterface;
+
+use function func_get_args;
+use function MongoDB\addLogger;
+use function MongoDB\removeLogger;
+
+/** @see https://jira.mongodb.org/browse/DRIVERS-2583 */
+class LogNonGenuineHostTest extends TestCase
+{
+    private LoggerInterface $logger;
+
+    public function setUp(): void
+    {
+        $this->logger = $this->createTestPsrLogger();
+
+        addLogger($this->logger);
+    }
+
+    public function tearDown(): void
+    {
+        removeLogger($this->logger);
+    }
+
+    /** @dataProvider provideCosmosUris */
+    public function testCosmosUriLogsInfoMessage(string $uri): void
+    {
+        $this->createClientAndIgnoreSrvLookupError($uri);
+
+        $expectedLog = [
+            'info',
+            'You appear to be connected to a CosmosDB cluster. For more information regarding feature compatibility and support please visit https://www.mongodb.com/supportability/cosmosdb',
+            ['domain' => 'mongoc'],
+        ];
+
+        $this->assertContains($expectedLog, $this->logger->logs);
+    }
+
+    public function provideCosmosUris(): array
```

Comment:
You should make the data provider static, as it will be required in PHPUnit 10+.

https://github.com/sebastianbergmann/phpunit/blob/20c23e85c86e5c06d63538ba464e8054f4744e62/ChangeLog-10.0.md?plain=1#L197-L198

Reply:
Fixed. Thanks for the reminder.